sycl/test-e2e/ESIMD/unified_memory_api/Inputs/block_store.hpp

```diff
@@ -219,6 +218,164 @@ bool testACC(queue Q, uint32_t Groups, uint32_t Threads,
+             slm_block_store(LocalElemOffset, Vals, StorePropertiesT{});
+           else
+             slm_block_store(LocalElemOffset, Vals);
+
+           Vals = slm_block_load<T, N>(LocalElemOffset);
+           Vals += 6;
+           slm_block_store(LocalElemOffset, Vals);
+           Vals = slm_block_load<T, N>(LocalElemOffset);
+         }
+         Vals.copy_to(OutPtr + ElemOff);
+       });
+     }).wait();
+  } catch (sycl::exception const &e) {
+    std::cout << "SYCL exception caught: " << e.what() << '\n';
+    return false;
+  }
+
+  bool Passed = verify(Out_val, Out.data(), Size, N, UseMask);
+
+  return Passed;
+}
+
+template <typename T, uint16_t N, bool UseMask, bool CheckProperties,
+          typename StorePropertiesT>
+bool testLocalAccSLM(queue Q, uint32_t Groups,
+                     StorePropertiesT StoreProperties) {
+  using shared_allocator = sycl::usm_allocator<T, sycl::usm::alloc::shared, 16>;
+  using shared_vector = std::vector<T, shared_allocator>;
+  constexpr uint16_t GroupSize = 8;
+
+  uint32_t Size = Groups * GroupSize * N;
+
+  std::cout << "Local Acc case: T=" << esimd_test::type_name<T>() << ",N=" << N
+            << ",UseMask=" << UseMask << ",UseProperties=" << CheckProperties
+            << std::endl;
+
+  sycl::range<1> GlobalRange{Groups};
+  sycl::range<1> LocalRange{GroupSize};
+  sycl::nd_range<1> Range{GlobalRange * LocalRange, LocalRange};
+
+  constexpr size_t Alignment = getAlignment<T, N, UseMask>(StoreProperties);
+
+  shared_vector Out(GroupSize, shared_allocator{Q});
+  T Out_val = esimd_test::getRandomValue<T>();
+  for (int i = 0; i < Size; i++)
+    Out[i] = Out_val;
+
+  try {
+    Q.submit([&](handler &CGH) {
+       local_accessor<T, 1> LocalAcc(GroupSize * N, CGH);
+
+       auto OutPtr = Out.data();
+
+       CGH.parallel_for(Range, [=](sycl::nd_item<1> ndi) SYCL_ESIMD_KERNEL {
+         constexpr uint32_t SLMSize = (GroupSize * N) * sizeof(T);
+         slm_init<SLMSize>();
+
+         uint16_t GlobalID = ndi.get_global_id(0);
+         uint16_t LocalID = ndi.get_local_id(0);
+         uint32_t LocalElemOffset = LocalID * N * sizeof(T);
+         uint32_t ElemOff = GlobalID * N;
+         simd<T, N> Vals(ElemOff, 1);
+
+         if constexpr (UseMask) {
+           simd_mask<1> Mask = (GlobalID + 1) & 0x1;
+           simd<T, N> PassThru = Out_val;
+           if constexpr (CheckProperties)
+             block_store(LocalAcc, LocalElemOffset, Vals, Mask,
+                         StorePropertiesT{});
+           else {
+             if (LocalElemOffset == 0)
+               block_store(LocalAcc, Vals, Mask);
+             else
+               block_store(LocalAcc, LocalElemOffset, Vals, Mask);
+           }
+           Vals = block_load<T, N>(LocalAcc, LocalElemOffset, Mask, PassThru);
+           Vals += 6;
+           block_store(LocalAcc, LocalElemOffset, Vals, Mask);
+           Vals = block_load<T, N>(LocalAcc, LocalElemOffset, Mask, PassThru);
```

Comment:
The test logic for masked block_store(local_accessor, ...) needs to be updated similarly to testSLM() function.